Makefile

```diff
@@ -0,0 +1,216 @@
+# Make does not offer a recursive wildcard function, so here's one:
+rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
+
+SHELL := /bin/bash
+NAME := jx-semanticcheck
+BINARY_NAME := jx-semanticcheck
+BUILD_TARGET = build
+MAIN_SRC_FILE=cmd/main.go
+GO := GO111MODULE=on go
+GO_NOMOD :=GO111MODULE=off go
+REV := $(shell git rev-parse --short HEAD 2> /dev/null || echo 'unknown')
+ORG := jenkins-x-plugins
+ORG_REPO := $(ORG)/$(NAME)
+RELEASE_ORG_REPO := $(ORG_REPO)
+ROOT_PACKAGE := github.com/$(ORG_REPO)
+GO_VERSION := 1.18
+GO_DEPENDENCIES := $(call rwildcard,pkg/,*.go) $(call rwildcard,cmd/,*.go)
+
+BRANCH     := $(shell git rev-parse --abbrev-ref HEAD 2> /dev/null  || echo 'unknown')
+BUILD_DATE := $(shell date +%Y%m%d-%H:%M:%S)
+CGO_ENABLED = 0
+
+REPORTS_DIR=$(BUILD_TARGET)/reports
+
+GOTEST := $(GO) test
+
+# set dev version unless VERSION is explicitly set via environment
+VERSION ?= $(shell echo "$$(git for-each-ref refs/tags/ --count=1 --sort=-version:refname --format='%(refname:short)' 2>/dev/null)-dev+$(REV)" | sed 's/^v//')
+
+# Build flags for setting build-specific configuration at build time - defaults to empty
+#BUILD_TIME_CONFIG_FLAGS ?= ""
+
+# Full build flags used when building binaries. Not used for test compilation/execution.
+BUILDFLAGS :=  -ldflags \
+  " -X $(ROOT_PACKAGE)/pkg/cmd/version.Version=$(VERSION)\
+		-X github.com/spring-financial-group/jx-semanticcheck/pkg/cmd/version.Version=$(VERSION)\
+		-X $(ROOT_PACKAGE)/pkg/cmd/version.Revision='$(REV)'\
+		-X $(ROOT_PACKAGE)/pkg/cmd/version.Branch='$(BRANCH)'\
+		-X $(ROOT_PACKAGE)/pkg/cmd/version.BuildDate='$(BUILD_DATE)'\
+		-X $(ROOT_PACKAGE)/pkg/cmd/version.GoVersion='$(GO_VERSION)'\
+		$(BUILD_TIME_CONFIG_FLAGS)"
+
+# Some tests expect default values for version.*, so just use the config package values there.
+TEST_BUILDFLAGS :=  -ldflags "$(BUILD_TIME_CONFIG_FLAGS)"
+
+ifdef DEBUG
+BUILDFLAGS := -gcflags "all=-N -l" $(BUILDFLAGS)
+endif
+
+ifdef PARALLEL_BUILDS
+BUILDFLAGS += -p $(PARALLEL_BUILDS)
+GOTEST += -p $(PARALLEL_BUILDS)
+else
+# -p 4 seems to work well for people
+GOTEST += -p 4
+endif
+
+ifdef DISABLE_TEST_CACHING
+GOTEST += -count=1
+endif
+
+TEST_PACKAGE ?= ./...
+COVER_OUT:=$(REPORTS_DIR)/cover.out
+COVERFLAGS=-coverprofile=$(COVER_OUT) --covermode=count --coverpkg=./...
+
+.PHONY: list
+list: ## List all make targets
+	@$(MAKE) -pRrn : -f $(MAKEFILE_LIST) 2>/dev/null | awk -v RS= -F: '/^# File/,/^# Finished Make data base/ {if ($$1 !~ "^[#.]") {print $$1}}' | egrep -v -e '^[^[:alnum:]]' -e '^$@$$' | sort
+
+.PHONY: help
+.DEFAULT_GOAL := help
+help:
+	@grep -h -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+full: check ## Build and run the tests
+check: build test ## Build and run the tests
+get-test-deps: ## Install test dependencies
```

Comment:
No need to use external dependencies, use standard libraries. See: https://github.com/jenkins-x-plugins/jx-project/pull/451 (It would be good to do this for other repos)